### PR TITLE
Implement use of email validation gem

### PR DESCRIPTION
**Pull request by KPobeeNorris**
It was established that we were duplicating the email address checking section of our codebase across PVB2 & PVB-public, and so this was extracted out into a rubygem.  This PR manages the implementation of the EmailAddressValidation gem.

### CODE REVIEW

File: spec/controllers/booking_requests_controller_spec.rb [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
   let(:visitors_details) {
     {
-      email_address: 'ada@test.example.com',
+      email_address: 'ada@test.example',
```
Comment (StupidCodeFactory) [inline spec/controllers/booking_requests_controller_spec.rb]: @KPobeeNorris why is the `.com` gone? It seems to me that `test.example.com` should pass the test.

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 require 'simplecov'
 require 'simplecov-rcov'
 SimpleCov.formatter = SimpleCov::Formatter::RcovFormatter
-SimpleCov.minimum_coverage 98.96
+SimpleCov.minimum_coverage 98.91
```
Comment (StupidCodeFactory) [inline spec/spec_helper.rb]: I'm creating a trello ticket to bump this up to 100%. We've got a good thing going having those at 100% coverage. Don't worry about this one, it's a bit trickier.

File: spec/controllers/booking_requests_controller_spec.rb (hunk 1 above)
Comment (KPobeeNorris) [reply spec/controllers/booking_requests_controller_spec.rb]: @StupidCodeFactory - thanks for the spot, I'll fix that now

File: spec/spec_helper.rb (hunk 2 above)
Comment (KPobeeNorris) [reply spec/spec_helper.rb]: Thanks @StupidCodeFactory

File: Gemfile [hunk 3]
```diff
@@ -29,6 +29,9 @@ gem 'pvb-instrumentation',
   git: 'https://github.com/ministryofjustice/pvb-instrumentation.git',
   ref: 'a264627211f2bf68f4e388200b2a050fe9081504'
 
+gem 'email_address_validation',
+  git: 'https://github.com/ministryofjustice/email_address_validation'
```
Comment (StupidCodeFactory) [inline Gemfile]: @KPobeeNorris just noticing this now. Can you create a tag like 'v0.1.0' and use point to this tag or at least reference the the commit using the `ref` keyword just like above.
That would ensure we use a deterministic version of the gem.

Comment (StupidCodeFactory) [reply Gemfile]: after that I think It's all good to go.

**Comment by KPobeeNorris** [comment]
@StupidCodeFactory - Implemented a reference in the Gemfile as requested, and made the same change in PVB2 too.

**Pull request merged**